crates/shared/src/bad_token/token_owner_finder/blockscout.rs

```diff
@@ -4,10 +4,15 @@ use prometheus::IntCounterVec;
 use prometheus_metric_storage::MetricStorage;
 use reqwest::{Client, Url};
 use serde::Deserialize;
+use std::time::Duration;
 
 use super::TokenOwnerFinding;
 
 const BASE: &str = "https://blockscout.com/";
+// Blockscout uses a custom timeout because their api is often slow. We would like those requests
+// to finish even if slow as bad token detection results are cached for a while and faster
+// TokenOwnerFinding implementations are not slowed down by slower ones.
+const TIMEOUT: Duration = Duration::from_secs(30);
```

Comment:
30s might not be long enough. I timed one request which caused a timeout alert earlier and it took 36 seconds.
Another one I tried timed out on their end after 45 seconds. So maybe this could be a good upper limit for us.